### PR TITLE
[testing] assert on fwd delcs with modules

### PR DESCRIPTION
**Pull request by Teemperor**

### CODE REVIEW

**Comment by Teemperor** [comment]
@phsft-bot just on slc6/gcc62 with flags -Druntime_cxxmodules=On -Dctest_test_exclude_none=on

**Comment by Teemperor** [comment]
@phsft-bot just on slc6/gcc62 with flags -Druntime_cxxmodules=On -Dctest_test_exclude_none=on

**Comment by Teemperor** [comment]
@phsft-bot build just on slc6/gcc62 with flags -Druntime_cxxmodules=On -Dctest_test_exclude_none=on

**Comment by Teemperor** [comment]
@phsft-bot build just on slc6/gcc62 with flags -Druntime_cxxmodules=On -Dctest_test_exclude_none=on

File: core/metacling/src/TCling.cxx [hunk 1]
```diff
@@ -1816,6 +1816,9 @@ void TCling::RegisterModule(const char* modulename,
    if (hasHeaderParsingOnDemand && fwdDeclsCode){
       // We now parse the forward declarations. All the classes are then modified
       // in order for them to have an external lexical storage.
+      if (fInterpreter->getCI()->getLangOpts().Modules) {
+        assert(false && "Fwd decls with modules?");
```
Comment (vgvassilev) [inline core/metacling/src/TCling.cxx]: We should probably put something like that in `TCling::ReadRootmapFile`.

**Pull request closed without merging**